tests/test_views.py

```diff
@@ -3256,6 +3256,26 @@ def test_register_email_mismatch(self):
         users = OSFUser.objects.filter(username=email)
         assert_equal(users.count(), 0)
 
+    def test_register_email_already_registered(self):
+        url = api_url_for('register_user')
+        name, email, password = fake.name(), fake_email(), fake.password()
+        existing_user = UserFactory(
+            username=email,
+        )
+        existing_user.save()
```

Comment:
Nit: `UserFactory()` implicitly saves, so this isn't necessary. Not a big deal though.